send/o365_mu_process.pl

```diff
@@ -229,11 +305,17 @@
 	print FILE_GROUPS_CACHE $groupRecord . "\n";
 }
 
+foreach my $resourceMailRecord (@allResourceMails) {
```

Comment:
Variable `@allResourceMails` is guarded by `$RESOURCE_MAILS_LOCK`. I'm aware that in this spot is the only one thread active. Therefore it's no need for the lock. But for the sake of future refactoring or changes in this script, I would put a lock here anyway or at least the comment explaining why the lock is not necessary.